File: Tree/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        queue=deque()
        ans=[]
        
        queue.append(root)
        while queue:
            n=len(queue)
            aux=[]
            for i in range(n):
                tem=queue.popleft()
                #print(tem.val)
                aux.append(tem.val)
                if tem.left:
                    queue.append(tem.left)
                if tem.right:
                    queue.append(tem.right)
            ans.append(aux)
        return ans    
                 
        

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        queue<TreeNode* > q;
        if(root==NULL){
            return {};
        }
        q.push(root);
        while(!q.empty()){
            int n=q.size();
            vector<int> temp;
            for(int i=0;i<n;i++){
                auto aux=q.front();
                temp.push_back(aux->val);
                q.pop();
                if(aux->left){
                    q.push(aux->left);
                }
                if(aux->right){
                    q.push(aux->right);
                }
                
            }
            ans.push_back(temp);
        }
        return ans;
    
        
    }
};